tests/microbenchmarks/attention_perf/attention_perf_harness.py

```
@@ -0,0 +1,354 @@
+
+
+# --------------------------------------------------------------------------- #
+# optional dependencies — graceful fallback (no hard requirement)
+# --------------------------------------------------------------------------- #
+
+
+def _try_clock_lock(gpu_id: int = 0):
+    """Return an active GPUClockLock or None. Locking kills DVFS (notes §9.4)."""
+    try:
+        from defs.perf.gpu_clock_lock import GPUClockLock  # type: ignore
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)